### DISCUSSION TITLE
Traffic lights in Hamburg: Moquette in action

### DISCUSSION THREAD

**Discussion by hylkevds** [moquette-io/moquette#719] [Show and tell]
**First, the demo.**
Live traffic light information from the city of Hamburg: [geoportal-hamburg.de](https://geoportal-hamburg.de/geo-online/?Map/layerIds=12883,12884,16101,19969,19968,23219,23214,23221,23216,23212,23210&Map/center=[565208,5935323]&Map/zoomLevel=9) (deep-link to the cool parts)

**Then some background.**
The city of Hamburg has a policy that all (non GDPR) geo-data they have should be open to the public, accessible using open standards. For publishing sensor data they choose the [OGC SensorThings API](https://docs.opengeospatial.org/is/18-088/18-088.html) standard, made by the [Open Geospatial Consortium](https://www.ogc.org/). The SensorThings API (STA) is a REST API for the typical CRUD actions, and it has MQTT bindings to allow for push notifications for any data changes.

One server implementation of STA is the open-source [FROST-Server](https://github.com/FraunhoferIOSB/FROST-Server), and this is also what the city of Hamburg is using for their sensor-data publication. For the MQTT implementation FROST uses an embedded Moquette instance, so the above demo is using Moquette to push live traffic light status using MQTT over websockets.

The total load is about 2000 Observations per second, with peaks of about 5000/second. The hard part is actually getting the data out of the traffic lights, since most traffic lights don't have a proper internet connection...

**Comment by andsel** [comment]
Thanks a ton @hylkevds for sharing this!! It's nice to see the project out in the wild!
